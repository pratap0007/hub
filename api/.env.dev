ENVIRONMENT="development"

POSTGRES_HOST="localhost"
POSTGRES_PORT="5432"
POSTGRES_DB="hub"
POSTGRES_USER="postgres"
POSTGRES_PASSWORD="postgres"

GH_CLIENT_ID=""
GH_CLIENT_SECRET=""
GHE_URL=""

GL_CLIENT_ID=""
GL_CLIENT_SECRET=""
GLE_URL=""

BB_CLIENT_ID=""
BB_CLIENT_SECRET=""

JWT_SIGNING_KEY=""
ACCESS_JWT_EXPIRES_IN=""
REFRESH_JWT_EXPIRES_IN=""

AUTH_BASE_URL=""

CONFIG_FILE_URL="file://../config.yaml"
