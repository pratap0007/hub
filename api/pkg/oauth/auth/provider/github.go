// Copyright Â© 2021 The Tekton Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package provider

import (
	"fmt"
	"os"

	"github.com/markbates/goth/providers/github"
)

type provider struct {
	Url          string
	ProfileUrl   string
	EmailUrl     string
	AuthUrl      string
	TokenUrl     string
	ClientId     string
	ClientSecret string
	CallbackUrl  string
}

func GithubProvider(AUTH_URL string) provider {
	githubAuth := provider{
		Url:          "https://github.com",
		ProfileUrl:   github.ProfileURL,
		EmailUrl:     github.EmailURL,
		ClientId:     os.Getenv("GH_CLIENT_ID"),
		ClientSecret: os.Getenv("GH_CLIENT_SECRET"),
		CallbackUrl:  fmt.Sprintf(AUTH_URL, "github"),
	}

	if os.Getenv("GHE_URL") != "" {
		githubAuth.Url = os.Getenv("GHE_URL")
		githubAuth.ProfileUrl = fmt.Sprintf("%s/api/v3/user", githubAuth.Url)
		githubAuth.EmailUrl = fmt.Sprintf("%s/api/v3/user/emails", githubAuth.Url)
	}

	githubAuth.AuthUrl = fmt.Sprintf("%s/login/oauth/authorize", githubAuth.Url)
	githubAuth.TokenUrl = fmt.Sprintf("%s/login/oauth/access_token", githubAuth.Url)

	return githubAuth
}
