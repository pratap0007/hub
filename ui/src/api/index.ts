import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource, IVersion } from '../store/resource';

interface Token {
  token: string;
  refreshInterval: string;
  expiresAt: number;
}

interface TokenData {
  access: Token;
  refresh: Token;
}

export interface AuthResponse {
  data: TokenData;
}

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
  resourceVersion(resourceId: number): Promise<IVersion>;
  versionUpdate(versionId: number): Promise<IVersion>;
  authentication(authCode: string): Promise<AuthResponse>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async authentication(authCode: string) {
    try {
      return axios.post(`${API_URL}/auth/login?code=${authCode}`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resourceVersion(resourceId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async versionUpdate(versionId: number) {
    try {
      return axios
        .get(`${API_URL}/resource/version/${versionId}`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }
}
