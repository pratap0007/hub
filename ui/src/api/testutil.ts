import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
