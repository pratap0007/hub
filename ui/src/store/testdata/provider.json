
  {
    "data": [
      {
        "name": "github"
      },
      {
        "name": "gitlab"
      },
      {
        "name": "bitbucket"
      }
    ]
  }


