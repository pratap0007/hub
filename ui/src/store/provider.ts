import { types, getEnv, flow, Instance } from 'mobx-state-tree';
import { Api } from '../api';

export const Provider = types.model('Provider', {
  name: types.identifier
});

export type IProvider = Instance<typeof Provider>;
export type IProviderStore = Instance<typeof ProviderStore>;

export const ProviderStore = types
  .model('ProviderStore', {
    items: types.map(Provider),
    isLoading: true,
    err: ''
  })

  .views((self) => ({
    get api(): Api {
      return getEnv(self).api;
    },

    get count() {
      return self.items.size;
    },

    get values() {
      return Array.from(self.items.values());
    }
  }))

  .actions((self) => ({
    add(item: IProvider) {
      self.items.put(item);
    },

    setLoading(l: boolean) {
      self.isLoading = l;
    }
  }))

  .actions((self) => ({
    load: flow(function* () {
      try {
        self.setLoading(true);
        const { api } = self;

        const json = yield api.providers();

        const providrs: IProvider[] = json.data.map((c: IProvider) => ({
          name: c.name
        }));

        providrs.forEach((c: IProvider) => self.add(c));
      } catch (err) {
        self.err = err.toString();
      }

      self.setLoading(false);
    })
  }))

  .actions((self) => ({
    afterCreate() {
      self.load();
    }
  }));
