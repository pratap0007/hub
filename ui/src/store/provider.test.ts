import { when } from 'mobx';
import { getSnapshot } from 'mobx-state-tree';
import { FakeHub } from '../api/testutil';
import { Provider, ProviderStore } from './provider';

const TESTDATA_DIR = `${__dirname}/testdata`;
const api = new FakeHub(TESTDATA_DIR);

describe('Store Object', () => {
  it('can create a provider object', () => {
    const provider = Provider.create({
      name: 'git'
    });

    expect(provider.name).toBe('git');
  });
});

describe('Store functions', () => {
  it('can create a Provider store', (done) => {
    const store = ProviderStore.create({}, { api });
    expect(store.count).toBe(0);
    expect(store.isLoading).toBe(true);
    when(
      () => !store.isLoading,
      () => {
        expect(store.count).toBe(3);
        expect(store.isLoading).toBe(false);

        expect(getSnapshot(store)).toMatchSnapshot();

        done();
      }
    );
  });
});
