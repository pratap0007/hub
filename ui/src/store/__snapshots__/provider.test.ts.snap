// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store functions can create a Provider store 1`] = `
Object {
  "err": "",
  "isLoading": false,
  "items": Object {
    "bitbucket": Object {
      "name": "bitbucket",
    },
    "github": Object {
      "name": "github",
    },
    "gitlab": Object {
      "name": "gitlab",
    },
  },
}
`;
